server/integrations/keycloak/initKeycloak.js

```diff
@@ -116,7 +118,8 @@ async function prepareKeycloak() {
 
       // Assign roles to the user
       await assignRealmRolesToUser(user, userIdUrl)
-      await assignClientRolesToUser(user, BEARER_CLIENT, userIdUrl)
+      await assignClientRolesToUser(user, BEARER_CLIENT, bearerClientRoles, userIdUrl)
+      await assignClientRolesToUser(user, PUBLIC_CLIENT, publicClientRoles, userIdUrl)
```

Comment:
basically the helper script used to assign some roles to the user only on the bearer client but because we use the public client in the backend also in the walkthrough etc, then the role checks are against the public client. This change makes sure the client is assigned the roles on both the public and bearer client so the role checks will work regardless of which client is used.

Reply:
[Question] Could we just export the realm json with this roles?

Reply:
Repeating question

Reply:
Yeah I'm pretty sure it's not possible. Any time I've tried before it didn't really work for me. This script is a little tricky to maintain because of all the handcrafted axios requests going to keycloak, but it could be replaced with a much simpler version of the same script we have in `keycloak-connect-graphql` https://github.com/aerogear/keycloak-connect-graphql/blob/master/scripts/initKeycloak.js